pkg/prometheus/promcfg.go

```diff
@@ -4154,6 +4154,57 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 		})
 	}
 
+	// IonosSDConfig
+	if len(sc.Spec.IonosSDConfigs) > 0 {
+		configs := make([][]yaml.MapItem, len(sc.Spec.IonosSDConfigs))
+		for i, config := range sc.Spec.IonosSDConfigs {
+			configs[i] = cg.addSafeAuthorizationToYaml(configs[i], s, &config.Authorization)
+			configs[i] = cg.addProxyConfigtoYaml(configs[i], s, config.ProxyConfig)
+
+			configs[i] = append(configs[i], yaml.MapItem{
+				Key:   "datacenter_id",
+				Value: config.DataCenterID,
+			})
+
+			if config.FollowRedirects != nil {
+				configs[i] = append(configs[i], yaml.MapItem{
+					Key:   "follow_redirects",
+					Value: config.FollowRedirects,
+				})
+			}
+
+			if config.EnableHTTP2 != nil {
+				configs[i] = append(configs[i], yaml.MapItem{
+					Key:   "enable_http2",
+					Value: config.EnableHTTP2,
+				})
+			}
+
+			if config.TLSConfig != nil {
+				configs[i] = cg.addSafeTLStoYaml(configs[i], s, config.TLSConfig)
+			}
```

Comment:
(nit) also we can move it next to authorization and proxy configs.
```suggestion
			configs[i] = cg.addSafeTLStoYaml(configs[i], s, config.TLSConfig)
```